core/page-assets/loader.php

```diff
@@ -123,12 +125,34 @@ public function enable_assets( array $assets_data ): void {
 				$this->assets[ $assets_type ][ $asset_name ]['enabled'] = true;
 
 				if ( 'scripts' === $assets_type ) {
+					if ( $this->should_import_script( $asset_name ) ) {
+						continue;
+					}
+
 					wp_enqueue_script( $asset_name );
 				} else {
 					wp_enqueue_style( $asset_name );
 				}
 			}
 		}
+
+		$this->list_import_scripts();
+	}
+
+	private function should_import_script( $script ): bool {
+		if ( false !== strpos( $script, 'import-script-' ) ) {
+			$script_name = str_replace( 'import-script-', '', $script );
+			$this->import_scripts[] = $script_name;
+			return true;
+		}
+
+		return false;
+	}
+
+	private function list_import_scripts(): void {
+		wp_register_script( 'scripts-import-list', '', [], 1.0 );
```

Comment:
@rami-elementor Is this a good way to create a JS script on the page or is there a better way?